source/client-side-encryption/tests/basic.template.yml

```diff
@@ -0,0 +1,42 @@
+database_name: &database_name "default"
+collection_name: &collection_name "default"
+
+encryption_setup:
+  collections:
+    - database: default
+      collection: default
+      data: []
+      jsonSchema: {{get_schema(0)}}
+    - database: admin
+      collection: datakeys
+      data: [{{get_key(0)}}]
+tests:
+  - description: "Insert a document requiring encryption" 
+    clientOptions:
+      clientSideEncryptionOptions:
+        kmsProvider: "aws-kms"
+        useRemoteSchemas: True
+    operations:
+      - name: insertOne
+        arguments:
+          document: &doc0 { _id: 1, ssn: "457-55-5462" }
+      - name: findOne
+        arguments:
+          filter: { _id: 1 }
+        result: *doc0
+    expectations:
+      - command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - &doc0_encrypted { _id: 1, ssn: {{get_ciphertext("457-55-5462", schema=0, field="ssn")}} }
+            ordered: true
+          command_name: "insert"
+      - command_started_event:
+          command:
+            find: *collection_name
+            filter: { _id: 1 }
+          command_name: find
+      - command_succeeded_event:
```

Comment:
Consider leaving out the succeeded events, for a few reasons

1. None of our other integration spec tests include them
2. If we include them for one of the commands, we have to include them for all.  Otherwise we'll have to specify complex matching rules in terms of what actual events to skip.
3. I think the intent here is to prove that the event is generated before decryption takes place.  If so, we can prove that in a single specialized test that only specifies succeeded events, or perhaps in a prose test.

Reply:
@jmikola also interested in what think about this one.

Reply:
It looks like the intention here is to assert that the server responds with encrypted data, while the operation's result assertion above expects unencrypted data. I wonder if we could accomplish this by using outcome assertions on the collection under test, if that was directed to use another MongoClient without FLE enabled.

I second Jeff's concerns about complications with deciding what commandSucceededEvents to skip. As-is, I think we'd need an assertion for the `insert` response to appear before this.

If my suggestion above about using outcome assertions isn't feasible, then I think a prose test would be warranted. That said, I wonder if the outcome assertions _not_ using FLE across the board would also help in other tests to assert that FLE properly encrypts application data before it hits server storage.

Reply:
I support the idea of an outcome assertion.  It's in line with other similar test runners, though in this case it is complicated by the need to use a different MongoClient.

Reply:
Based on @kevinAlbs last comment, it doesn't seem like the outcome assertion or commandSucceededEvent are necessary here. If commandStartedEvent for `insert` contains encrypted data, we can trust that is what is being stored server-side. Do we still want a prose test to verify a server response with encrypted data by observing commandSucceededEvent?

Reply:
I'm adding back "outcome". It's actually a bit easier to verify that data is stored as ciphertext now that auto encryption is set on the collection level. Operations that require auto encryption pass that as a collection option. The "outcome" data will be run on a collection without auto encryption/decryption, so it should see the ciphertext.